Moda.Web/src/moda.web.reactclient/src/app/settings/work-management/work-type-slice.ts

```diff
@@ -0,0 +1,53 @@
+import { WorkTypeDto } from '@/src/services/moda-api'
+import { getWorkTypesClient } from '@/src/services/clients'
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+
+interface WorkTypeState {
+  isLoading: boolean
+  workTypes: WorkTypeDto[]
+  includeInactive: boolean
+  error: any | null
+}
+
+const initialState: WorkTypeState = {
+  isLoading: false,
+  workTypes: [],
+  includeInactive: false,
+  error: null,
+}
+
+// Generates pending, fulfilled, and rejected action types
+export const fetchWorkTypes = createAsyncThunk(
+  'workType/fetchWorkTypes',
+  async (includeInactive: boolean) => {
+    return await (await getWorkTypesClient()).getList(includeInactive)
+  },
+)
+
+const workTypeSlice = createSlice({
+  name: 'workType',
+  initialState: initialState,
+  reducers: {
+    setIncludeInactive: (state, action) => {
+      state.includeInactive = action.payload
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(fetchWorkTypes.pending, (state) => {
+      state.isLoading = true
```

Comment:
I think best practice is to set the error state to null here in the pending action vs the fulfilled action.

Reply:
Updated.